test/index/types/types.snapshot.cc

```diff
@@ -143,3 +155,31 @@
   #undef QUALIFIED
 //       ^^^^^^^^^ reference [..] `types.cc:90:9`!
   }
+  
+  /// Restating what's already implied by the name
+  class DocumentedForwardDeclaration;
+//      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ reference [..] DocumentedForwardDeclaration#
+  
+  class DocumentedForwardDeclaration { };
+//      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ definition [..] DocumentedForwardDeclaration#
+//      documentation
+//      | Restating what's already implied by the name
+  
+  class Parent {};
+//      ^^^^^^ definition [..] Parent#
+  
+  class Child: Parent {};
+//      ^^^^^ definition [..] Child#
+//             ^^^^^^ reference [..] Parent#
+  
+  template <class CRTPChild>
+  class CRTPBase {
+//      ^^^^^^^^ definition [..] CRTPBase#
+    void castAndDoStuff() { static_cast<CRTPChild *>(this)->doStuff(); }
+  };
+  
+  class CRTPChild: CRTPBase<CRTPChild> {
```

Comment:
TODO: We should fix the missing reference for CRTPBase here.

Reply:
Hmm, the most obvious way that I tried this doesn't seem to work. Instead of blocking on this, let's punt adding support for templates to later, which will include handling both definitions and references properly.